client/batch.go

```diff
@@ -116,24 +116,27 @@ func (b *Batch) fillResults() error {
 				row := &result.Rows[k]
 				row.Key = []byte(call.Args.(*proto.IncrementRequest).Key)
 				if result.Err == nil {
-					// TODO(pmattis): This is odd. KeyValue.Value is otherwise always a
-					// []byte except for setting it to a *int64 here.
-					row.Value = &t.NewValue
+					row.Value = &proto.Value{
+						Bytes: encoding.EncodeUint64(nil, uint64(t.NewValue)),
+						Tag:   proto.ValueType_INT,
+					}
 					row.setTimestamp(t.Timestamp)
 				}
 			case *proto.ScanResponse:
 				result.Rows = make([]KeyValue, len(t.Rows))
-				for j, kv := range t.Rows {
-					row := &result.Rows[j]
-					row.Key = kv.Key
-					row.setValue(&kv.Value)
+				for j := range t.Rows {
+					src := &t.Rows[j]
```

Comment:
It's meaningful. We're now keeping a pointer to `&t.Rows[j].Value` in `dst`. We can't use `&kv.Value` because `kv` was a temporary variable.